File: CS50X Computer Science/sentimental-mario-more/mario.py
""" Write, in a file called mario.py, a program that recreates these half-pyramids using hashes (#) for blocks, exactly as you did in Problem Set 1, except that your program this time should be written in Python.
To make things more interesting, first prompt the user with get_int for the half-pyramid’s height, a positive integer between 1 and 8, inclusive. (The height of the half-pyramids pictured above happens to be 4, the width of each half-pyramid 4, with a gap of size 2 separating them).
If the user fails to provide a positive integer no greater than 8, you should re-prompt for the same again.
Then, generate (with the help of print and one or more loops) the desired half-pyramids.
Take care to align the bottom-left corner of your pyramid with the left-hand edge of your terminal window, and ensure that there are two spaces between the two pyramids, and that there are no additional spaces after the last set of hashes on each row. """


import cs50

while True:
	height = cs50.get_int("Choose the height of your pyramid from 1 to 8\n")
	if 1 <= height <= 8:
		break


for i in range (1, height + 1):
	spaces = ' ' * (height - i)
	blocks = '#' * i
	middle = '  '
	print(f"{spaces}{blocks}{middle}{blocks}")

File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

// Function to print spaces
void print_spaces(int count)
{
    for (int i = 0; i < count; i++)
    {
        printf(" ");
    }
}

// Function to print hashes
void print_hashes(int count)
{
    for (int i = 0; i < count; i++)
    {
        printf("#");
    }
}

int main(void)
{
    int height = 0;
    do
    {
        height = get_int("Choose a height between 1 and 8: ");
    }
    while (height < 1 || height > 8);

    for (int i = 1; i <= height; i++)
    {
        // Print spaces for left pyramid
        print_spaces(height - i);

        // Print hashes for left pyramid
        print_hashes(i);

        // Print gap between pyramids
        printf("  ");

        // Print hashes for right pyramid
        print_hashes(i);

        printf("\n");
    }
}
